prusti-common/src/vir/ast/expr.rs

```diff
@@ -1631,6 +1631,18 @@ impl Expr {
         let mut patcher = TypePatcher { substs: substs };
         patcher.fold(self)
     }
+
+    /// Is this expression a constant?
+    pub fn is_constant(&self) -> bool {
+        match self {
+            Expr::Const(_, _) => true,
+            Expr::UnaryOp(_, box subexpr, _) => subexpr.is_constant(),
+            Expr::BinOp(_, box subexpr1, box subexpr2, _) => {
+                subexpr1.is_constant() && subexpr2.is_constant()
+            }
+            _ => false,
```

Comment:
What is this for? You might want to add (domain) function applications with only constant arguments. One example that comes to mind is the "unit domain" produced as the snapshot for data-less types.

Reply:
Good point. However, I would prefer in this PR to touch VIR as little as possible.